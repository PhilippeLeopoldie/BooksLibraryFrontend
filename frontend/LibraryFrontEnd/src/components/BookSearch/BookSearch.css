.search_form--input {
  margin-top: 2rem;
  font-size: 1.2rem;
  width: 19rem;
}

.search_form--input--black {
    color: var(--fontcolor-dark);
    border: 1px var(--forminputcolor) solid;
}

.search_form--black {
        background-color: inherit;
}

