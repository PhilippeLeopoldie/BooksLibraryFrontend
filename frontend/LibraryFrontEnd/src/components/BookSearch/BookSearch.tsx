import { FormEvent, useContext, useState } from "react";
import "./BookSearch.css";
import { BookSearchList } from "./BookSearchList/BookSearchList";
import { ThemeContext } from "../../App/App";


export const BookSearch = () => {
  const theme = useContext(ThemeContext);
  const [searchCriteria, setSearchCriteria] = useState<string>("");

  const handleSearch = (e : FormEvent) => {
    e.preventDefault();
    const inputField = document.activeElement as HTMLElement;

    // remove focus from text input
    if(inputField){ inputField.blur() };  
  }

  return (
    <>
      <form
        className={"search_form--" + theme}
        onSubmit={handleSearch}
      >
          <input
          type="search"
          className={`search_form--input search_form--input--${theme}`}
          autoFocus
          placeholder="Search by Title or Author..."
          name="titleOrAuthor"
          value={searchCriteria}
          onChange={(e) => {
            setSearchCriteria(e.target.value.toLowerCase());
          }}
          onKeyDown={(e) => {
            if (e.key === "Enter") e.preventDefault();
          }}
        />
        <BookSearchList titleOrAuthor={searchCriteria} />
      </form>
    </>
  );
};
