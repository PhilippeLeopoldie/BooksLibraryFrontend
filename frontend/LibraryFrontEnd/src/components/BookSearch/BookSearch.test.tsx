import { BookSearch } from "./BookSearch";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { useState } from "react";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: jest.fn(),
}));

type BooksSearchCriteria = {
  search?:{
    title: string,
    author: string
  }
}

const book : BooksSearchCriteria = {
  search:{
    title: "",
    author: ""
  }
}

describe("BookList rendering", () => {
  test("snapShot: should render BookList", () => {
    // Arrange + Act
    (useState as jest.Mock).mockImplementation(() => [book, jest.fn()]);
    (useState as jest.Mock).mockImplementation(() => ["author", jest.fn()]);
    const {asFragment} = render(<BookSearch/>)

    // Expect
    expect(asFragment()).toMatchSnapshot();
  })

  test("snapShot: should render an empty BookList", () => {
    // Arrange + Act
    (useState as jest.Mock).mockImplementation(() => [book, jest.fn()]);
    (useState as jest.Mock).mockImplementation(() => ["author", jest.fn()]);
    const {asFragment} = render(<BookSearch/>)

    // Expect
    expect(asFragment()).toMatchSnapshot();
  })
})