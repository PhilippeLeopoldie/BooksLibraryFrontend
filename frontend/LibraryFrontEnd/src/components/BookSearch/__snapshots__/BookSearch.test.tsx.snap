// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList rendering snapShot: should render BookList 1`] = `
<DocumentFragment>
  <form
    class="search-form--natural"
  >
    <script>
      #f3f3f4
    </script>
    <input
      class="search-form__input"
      name="titleOrAuthor"
      placeholder="Search by Title or Author"
      value="author"
    />
    <h1
      class="bookListSearchContainer--natural"
    >
      author
    </h1>
  </form>
</DocumentFragment>
`;

exports[`BookList rendering snapShot: should render an empty BookList 1`] = `
<DocumentFragment>
  <form
    class="search-form--natural"
  >
    <script>
      #f3f3f4
    </script>
    <input
      class="search-form__input"
      name="titleOrAuthor"
      placeholder="Search by Title or Author"
      value="author"
    />
    <h1
      class="bookListSearchContainer--natural"
    >
      author
    </h1>
  </form>
</DocumentFragment>
`;
